src/Compilers/Core/Portable/Serialization/StreamObjectWriter.cs

```diff
@@ -1506,7 +1544,7 @@ public static Variant FromUInt32(uint value)
 
             public static Variant FromUInt64(ulong value)
             {
-                return new Variant(VariantKind.UInt64, value);
+                return new Variant(VariantKind.UInt64, unchecked((long)value));
```

Comment:
ulongs can be safely roundtripped using a long.  no bits are lost doing this.  Tests have been added to ensure roundtripping works properly here.